daemon/src/main/scala/co/ledger/wallet/daemon/models/Account.scala

```
@@ -46,18 +48,20 @@ object Account {
       } yield AccountView(wallet.name, index, ba, ops, coreA.getRestoreKey, wallet.currency.currencyView)
     }
 
+    // TODO: This part only works for BTC, need a way to scale it to all coins
     def signTransaction(rawTx: Array[Byte], signatures: Seq[(Array[Byte], Array[Byte])]): Future[String] = {
-      val tx = wallet.currency.parseUnsignedTransaction(rawTx)
+      // TODO avoid brute force Either resolve
+      val tx = wallet.currency.parseUnsignedBTCTransaction(rawTx).right.get
```

this exception is not handled